### PR TITLE
ESP8285 updates

### PR DESCRIPTION
**Pull request by olliw42**
this PR does a couple of things. It is - AFAICS - the latest and greatest for ESP8285

- sx spi api changed to use transferBytes for ESP
- spi uses low level in select/deselect as suggested by @jlpoltrack, and found to be faster
- updates submodules, reverts back to using mLRS sx submodule
- tidies esp hal
- tidies esp lib
- syncs to main repo

tested on SpeedyBee Nano 2.4G receiver (generic-2400-pa)

### CODE REVIEW

File: mLRS/Common/hal/esp/rx-hal-bayckrc-900-pa-esp8285.h [hunk 1]
```diff
@@ -2,106 +2,94 @@
...
 
 
 //-- SX1: SX12xx & SPI
-#define SPI_CS_IO                 15
-#define SPI_FREQUENCY             10000000L
-#define SX_RESET                  2
-#define SX_DIO0                   4
-#define SX_DIO1                   5
+#define SPI_CS_IO                 IO_P15
+#define SPI_FREQUENCY             16000000L
```
Comment (jlpoltrack) [inline mLRS/Common/hal/esp/rx-hal-bayckrc-900-pa-esp8285.h]: Friendly note - SX127x SPI should run at 10 MHz.  Only SX128x is rated for 16 MHz.

File: mLRS/Common/hal/esp/rx-hal-generic-900-esp8285.h [hunk 2]
```diff
@@ -2,106 +2,94 @@
...
 
 
 //-- SX1: SX12xx & SPI
-#define SPI_CS_IO                 15
-#define SPI_FREQUENCY             10000000L
-#define SX_RESET                  2
-#define SX_DIO0                   4
-#define SX_DIO1                   5
+#define SPI_CS_IO                 IO_P15
+#define SPI_FREQUENCY             16000000L
```
Comment (jlpoltrack) [inline mLRS/Common/hal/esp/rx-hal-generic-900-esp8285.h]: Same as note above - should be 10 MHz.

Comment (olliw42) [reply mLRS/Common/hal/esp/rx-hal-generic-900-esp8285.h]: THX! done.

File: mLRS/Common/hal/esp/rx-hal-bayckrc-900-pa-esp8285.h (hunk 1 above)
Comment (olliw42) [reply mLRS/Common/hal/esp/rx-hal-bayckrc-900-pa-esp8285.h]: THX! done.

**Comment by tmcadam** [comment]
This looks good. I have tested on generic_900 and e77_mbl_900 boards. I like the approach of using the generic hal and inheritance in platformio.ini to build named targets for specific hardware.

**Pull request merged**